core-rust/state-manager/src/store/in_memory.rs

```diff
@@ -103,8 +104,9 @@ impl InMemoryStore {
     pub fn new() -> InMemoryStore {
         InMemoryStore {
             transactions: BTreeMap::new(),
-            transaction_receipts: BTreeMap::new(),
             transaction_identifiers: BTreeMap::new(),
+            ledger_receipts: BTreeMap::new(),
```

Comment:
I wonder if it's actually worth it to store these separately or whether it just introduces overhead?

I guess storing them separately allows us to possibly prune them separately easily? 👍

Reply:
The main factor for splitting at DB came from Łukasz (https://github.com/radixdlt/babylon-node/pull/372#discussion_r1143667148): I think his suggested use-case was that we could have an app flag driving whether an instance should store the "local execution" at all or not (I guess it touches your "full node vs validator").

The DB code change wasn't too cumbersome and the overhead doesn't seem crazy ("do 2 lookups instead of 1"), so I went with it.

One more minor advantage: in some places in the code I've seen that (after some extra changes _not_ belonging in this PR) we could load just the smaller LedgerTransactionReceipt.

Generally, my opinion here is not very strong. If the DB-level split causes any problems, I can easily revert this part.

Reply:
> [sometimes] we could load just the smaller LedgerTransactionReceipt

...ah, and potentially more of such places will pop up after https://github.com/radixdlt/babylon-node/pull/372#discussion_r1146167124 happens.